src/server/gc_worker.rs

```diff
@@ -1223,20 +1204,38 @@ impl<E: Engine> GCWorker<E> {
         Ok(())
     }
 
+    /// Check whether GCWorker is busy. If busy, callback will be invoked with an error that
+    /// indicates GCWorker is busy; otherwise, return a new callback that invokes the original
+    /// callback as well as decrease the scheduled task counter.
+    fn check_is_busy<T: 'static>(&self, callback: Callback<T>) -> Option<Callback<T>> {
```

Comment:
How about implementing it in `FutureWorker`?

Reply:
I'm afraid that too many code will be changed since currently the only possible error of `FutureScheduler::schedule` is `Stopped`, and now we are adding another type of error. Maybe later we can refine it.